aten/src/ATen/TensorIterator.cpp

```diff
@@ -1496,7 +1496,8 @@ void TensorIteratorBase::build(TensorIteratorConfig& config) {
       common_device_.type() == DeviceType::IPU  ||
       common_device_.type() == DeviceType::Lazy ||
       common_device_.type() == DeviceType::ORT  ||
-      common_device_.type() == DeviceType::HPU) return;
+      common_device_.type() == DeviceType::HPU  ||
+      common_device_.type() == DeviceType::PrivateUse1) return;
```

Comment:
Hmm was something breaking that required you to add `PrivateUse1` to the list here? It seems suspicious to add it here, because `PrivateUse1` has no guarantees around its storage

I actually don't think that we want this here, because it implies that the `PrivateUse1` device is

Reply:
This makes sense. The iteration of an on-device tensor can always be done "on the device"; on the host we can just convert it back to CPU tensor first. Will remove this.